### PR TITLE
Update readme & contributing ToC icon

### PR DESCRIPTION
**Pull request by janbrasna**
Ever since the icon/UI changed to more legible ToC icon (or its size) this little snippet has been bugging me. It has bad drop shadow that's cropped, is not aligned and today doesn't reflect the actual UI affordance.

### Why:

1. The icon as captured in the past doesn't reflect today's UI.
2. The asset doesn't have square aspect ratio however the readmes apply 25×25px size on the img element making the png rendered distorted.
3. It doesn't work well with dark themes…

### What's being changed (if available, include any code snippets, screenshots, or gifs):
 
`./contributing/images/table-of-contents.png` occurence:
1. `./README.md`
2. `.github/CONTRIBUTING.md` (updated to add the same alt text from ./README)

Screen Shot 2024-03-02 at 22 20 12

PS: The updated PNG icon might not display straight away as images are being cached heavily; so to see the change rendered inside the markdown previews in the repo some sha/ref trickery might be needed to view the tree URL not previously visited to get the changed asset shown in the MDs as well.

PPS: The image has been pngoptimized to ~1kB, no more work needed.

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Comment by nguyenalex836** [comment]
@janbrasna Thanks so much for opening a PR with this fix! I'll update the branch and get this merged once tests are passing 🍏

**Review by nguyenalex836 — APPROVED** [review]

File: contributing/images/table-of-contents.png
Comment (felix30497989) [inline contributing/images/table-of-contents.png:1]: contributing/images/table-of-contents.png

**Review by felix30497989 — COMMENTED** [review]
Hjil``

**Review by felix30497989 — COMMENTED** [review]
Hjil``

**Comment by felix30497989** [comment]
> Hjil``

https://github.com/github/docs/pull/31928#pullrequestreview-1919550142

**Comment by felix30497989** [comment]
github/docs/tsconfig.json

**Pull request merged**